iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/evaluator/CompactionEvaluator.java

```diff
@@ -0,0 +1,118 @@
+
+public class CompactionEvaluator {
+
+  private CompactionEvaluator() {
+
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(CompactionEvaluator.class);
+
+  public static boolean isEligibleForCompaction(Table icebergTable, String partitionPath,
+      CompactionType compactionType, HiveConf conf) {
+
+    if (icebergTable.currentSnapshot() == null) {
+      LOG.info("Table {}{} doesn't require compaction because it is empty", icebergTable,
+          partitionPath == null ? "" : " partition " + partitionPath);
+      return false;
+    }
+
+    CommonPartitionEvaluator partitionEvaluator = constructCommonPartitionEvaluator(icebergTable, partitionPath, conf);
+
+    switch (compactionType) {
+      case MINOR:
+        return partitionEvaluator.isMinorNecessary();
+      case MAJOR:
+        return partitionEvaluator.isFullNecessary() || partitionEvaluator.isMajorNecessary();
+      default:
+        return false;
+    }
+  }
+
+  private static TableRuntime constructTableRuntime(Table icebergTable, HiveConf conf) {
+    long fileSizeInBytesThreshold = HiveConf.getSizeVar(conf,
+        HiveConf.ConfVars.HIVE_ICEBERG_COMPACTION_FILE_SIZE_THRESHOLD);
+
+    OptimizingConfig optimizingConfig = OptimizingConfig.parse(Collections.emptyMap());
+    optimizingConfig.setTargetSize(fileSizeInBytesThreshold);
+    optimizingConfig.setFullTriggerInterval(0);
+    optimizingConfig.setMinorLeastInterval(0);
+
+    TableConfiguration tableConfig = new TableConfiguration();
+    tableConfig.setOptimizingConfig(optimizingConfig);
+
+    TableRuntimeMeta tableRuntimeMeta = new TableRuntimeMeta();
+    tableRuntimeMeta.setTableName(icebergTable.name());
+    tableRuntimeMeta.setFormat(TableFormat.ICEBERG);
+    tableRuntimeMeta.setLastFullOptimizingTime(1);
+    tableRuntimeMeta.setLastMinorOptimizingTime(1);
+    tableRuntimeMeta.setTableConfig(tableConfig);
+
+    return new HiveTableRuntime(tableRuntimeMeta);
+  }
+
+  private static CommonPartitionEvaluator constructCommonPartitionEvaluator(Table table, String partitionPath,
+      HiveConf conf) {
+    TableRuntime tableRuntime = constructTableRuntime(table, conf);
+
+    TableFileScanHelper tableFileScanHelper = new IcebergTableFileScanHelper(table,
+        table.currentSnapshot().snapshotId());
+    CommonPartitionEvaluator evaluator = null;
+    try (CloseableIterable<TableFileScanHelper.FileScanResult> results =
+             tableFileScanHelper.scan()) {
+      for (TableFileScanHelper.FileScanResult fileScanResult : results) {
+        DataFile file = fileScanResult.file();
+        if (IcebergCompactionUtil.doesFileMatchPartition(table, partitionPath, file)) {
+          PartitionSpec partitionSpec = table.specs().get(file.specId());
+          Pair<Integer, StructLike> partition = Pair.of(partitionSpec.specId(), fileScanResult.file().partition());
+          evaluator = evaluator == null ? new CommonPartitionEvaluator(tableRuntime, partition, 2) : evaluator;
```

Comment:
what. is the magic number `2`?

Reply:
The number `2` is a hardcoded plan time at which we run compaction evaluator.
Amoro compaction evaluator has logic that uses last full optimize time, last minor optimize time and the current plan time. I defined last full optimize time and last minor optimize as `1`. With these hardcoded values the time-based conditions in Amoro compaction evaluator are always true and the results of the evaluation depends on other metrics.

Reply:
would it make sence to use `System.currentTimeMillis()` everywhere istead of this constants?

Reply:
I think `System.currentTimeMillis()` makes sense for the current plan time, but for last optimize time we can use something hardcoded, because last optimize time was not done at `System.currentTimeMillis()` time.

if we use `System.currentTimeMillis()` as the last optimize time then it will be same as current plan time which is incorrect.